File: reto18/reto18.py
def find_in_agenda(agenda: str, phone: str) -> dict | None:
    infos = agenda.split("\n")
    phone_infos = [info for info in infos if info.includes(phone)]
    if len(phone_infos) != 1 or len(phone_infos) > 1:
        return None
    info = phone_infos[0]
    return {
        "name": info[info.index("<") + 1 : info.index(">")],
        "address": info.replace("<.*>", "")
        .replace("+[0-9]{1,2}-[0-9]{3}-[0-9]{3}-[0-9]{3}", "")
        .strip(),
    }

File: reto18/reto18.ts
function findInAgenda(
  agenda: string,
  phone: string
): { name: string; address: string } | null {
  const infos = agenda.split("\n");
  const phoneInfos = infos.filter((info) => info.includes(phone));
  if (phoneInfos.length !== 1 || phoneInfos.length > 1) return null;
  const info = phoneInfos[0];
  return {
    name: info.slice(info.indexOf("<") + 1, info.indexOf(">")),
    address: info
      .replace(/\<.*\>/g, "")
      .replace(/\+[0-9]{1,2}-[0-9]{3}-[0-9]{3}-[0-9]{3}/g, "")
      .trim(),
  };
}
